src/backend/Donate/Actions/DonateForm/GetWarehouses/GetWarehousesService.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Core.Database;
+using Core.Exceptions;
+using Core.Services;
+using HashidsNet;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using System.Linq.Dynamic.Core;
+using Z.EntityFramework.Plus;
+
+namespace Donate.Actions.DonateForm.GetWarehouses
+{
+    public class GetWarehousesService
+    {
+        private readonly AppDbContext _appDbContext;
+        private readonly ILogger<GetWarehousesService> _logger;
+        private readonly Hashids _hashids;
+
+        public GetWarehousesService(AppDbContext appDbContext, ILogger<GetWarehousesService> logger, Hashids hashids)
+        {
+            _appDbContext = appDbContext;
+            _logger = logger;
+            _hashids = hashids;
+        }
+
+        public async Task<List<GetWarehousesResponse>> GetActiveWarehouses()
+        {
+            var listOfWarehouses = await _appDbContext.Warehouses
+                .Include(c => c.Address).ThenInclude(c => c!.Country)
+                .Where(c => c.Status == Core.Database.Enums.DbEntityStatus.Active).FromCache().ToDynamicListAsync();
+
+            if (listOfWarehouses.Count == 0)
```

Comment:
`!listOfWarehouses.Any()`, bo aktualnie .NET pod spodem leci pętlą i liczy ilość wystąpień (może np tak lecieć 10.000 razy, bo taka duża kolekcja). Jak dasz Any() to polepszasz performance.